.devcontainer/Dockerfile

```diff
@@ -0,0 +1,111 @@
+###############################################################################
+# Use a specific ubuntu version so we're not suprised by silent changes to gcc
+# versions (and thus C++ feature support). Jammy comes with gcc 11:
+# https://packages.ubuntu.com/jammy/gcc
+FROM ubuntu:jammy-20230804
+
+# Register relevant parameters for user creation.
+ARG UNAME=vscode
+ARG UID=1000
+ARG GID=1000
+
+# Installs `wget` and `curl` to aid debugging and to install other tools, with
+# ca-certificates to make https calls.
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    ca-certificates \
+    wget \
+    curl \
+    && rm -rf /var/lib/apt/lists/*
+
+ARG CLANG_VERSION=14
+
+RUN apt-get update \
+    # Install various prerequisite packages need for building as well as
+    # packages that aid developing and debugging.
+    #
+    # As the list/version of packages is not changing too often, we optimize
+    # for image layer size and run all `apt` related commands (that includes
+    # the invokation of `llvm.sh`) in one `RUN` statement.
+    #
+    # If you are adding packages to the list below, please add a comment here
+    # as well.
+    #
+    # Additional mandatory packages we install:
+    #  * autoconf: required to build cc_image targets (alexmc: I think!).
+    #  * build-essential: get gcc and std headers.
+    #  * ca-certificates: dependency for curl to make https calls.
+    #  * curl: not strictly needed (while-false: I think) outside of image
+    #    building but small (~100kb) and useful for debugging. Might be used
+    #    internally by bazel to fetch `http_archives`.
+    #  * gnupg: for image and package signing.
+    #  * git: Used by `bazel` to fetch a `git_repository`.
+    #  * lsb-release: to allow install scripts/(debugging )developers to figure
+    #    out what system they are on.
+    #  * make: required to build cc_image targets (alexmc: I think!).
+    #  * wget: see `curl`.
+    #
+    && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+    # Mandatory packages:
+    autoconf \
+    build-essential \
+    gnupg \
+    git \
+    lsb-release \
+    software-properties-common \
+    zlib1g-dev \
```

Comment:
Did you add this to the list? Then please also add it to the comment above to explain why.